tests/udp/test_environment.rs

```diff
@@ -0,0 +1,100 @@
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use torrust_tracker::protocol::info_hash::InfoHash;
+use torrust_tracker::tracker::peer::Peer;
+use torrust_tracker::tracker::Tracker;
+use torrust_tracker::udp::server::{RunningUdpServer, StoppedUdpServer, UdpServer};
+
+use crate::common::tracker::new_tracker;
+
+#[allow(clippy::module_name_repetitions, dead_code)]
+pub type StoppedTestEnvironment = TestEnvironment<Stopped>;
+#[allow(clippy::module_name_repetitions)]
+pub type RunningTestEnvironment = TestEnvironment<Running>;
+
+pub struct TestEnvironment<S> {
+    pub cfg: Arc<torrust_tracker_configuration::Configuration>,
+    pub tracker: Arc<Tracker>,
+    pub state: S,
+}
+
+#[allow(dead_code)]
+pub struct Stopped {
+    udp_server: StoppedUdpServer,
+}
+
+pub struct Running {
+    udp_server: RunningUdpServer,
+}
+
+impl<S> TestEnvironment<S> {
+    /// Add a torrent to the tracker
+    #[allow(dead_code)]
+    pub async fn add_torrent(&self, info_hash: &InfoHash, peer: &Peer) {
+        self.tracker.update_torrent_with_peer_and_get_stats(info_hash, peer).await;
+    }
+}
```

Comment:
Hi @WarmBeer, this is OK but if you read [this discussion](https://github.com/torrust/torrust-tracker/discussions/189) I think we should stop using that method. In fact, it should be private.

Besides, now we have a high-level public method to announce a new peer: `Tracker::announce`. Maybe in this case we could use:

```rust
impl<S> TestEnvironment<S> {
    /// Add a torrent to the tracker
    #[allow(dead_code)]
    pub async fn announce_torrent(&self, info_hash: &InfoHash, peer: &mut Peer, remote_client_ip: &IpAddr) {
        self.tracker.announce(info_hash, peer: &mut Peer, remote_client_ip).await;
    }
}
```

Althougth that could have side effects we do not want in these tests.